src/session_handle.cpp

```diff
@@ -760,12 +760,14 @@ namespace {
 
 	unsigned short session_handle::listen_port() const
 	{
-		return sync_call_ret<unsigned short>(&session_impl::listen_port);
+		return sync_call_ret<unsigned short, unsigned short(session_impl::*)() const>
```

Comment:
I guess you could use the local endpoint we expose in `announce_endpoint` but that would be subject to race conditions.